src/client.py

```diff
@@ -0,0 +1,362 @@
+        :rtype: dict, None
+        """
+        if not showall:
+            return self.api_root('projects')
+        else:
+            payload = 'showall'
+            return self.api_root('projects', parameters=payload)
+
+    def get_my_project(self, name=None, id=None):
+        """
+        Gets a specific MERMAID project. Either name or id required.
+        See https://mermaid-api.readthedocs.io/en/latest/projects.html for more information on project resources.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :return: project with associated data.
+        :rtype: dict, None
+        """
+        if id:
+            return self.api_projects(id=id)
+        elif name:
+            projects = self.get_projects(showall=True)
+            # Projects not None type.
+            if projects:
+                return lookup('name', name, projects['results'])
+        return None
+
+    def get_project_id(self, name=None, project=None):
+        """
+        Gets an ID for given MERMAID project.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type: str
+        :param project: (optional if 'name' provided) project dict containing project data.
+        eg. project dict returned from get_my_project().
+        :type: dict
+        :return: project ID.
+        :rtype: str, None
+        """
+        if project and isinstance(project, dict):
+            return project.get('id')
+        elif name:
+            project = self.get_my_project(name)
+            # Nested if instead of 'elif name and self.get_my_project(name=name)' to improve performance by reducing
+            # the number of API calls for name or id NoneType check.
+            if project:
+                return project.get('id')
+        return None
+
+    def get_project_data(self, data, name=None, id=None):
+        """
+        Gets project resource data including resources and observations. Authenticated access to project data
+        depends on the association of a user profile with a project. Unauthenticated access to project data
+        depends on the data sharing policies chosen per survey method for a project.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :param data: project resource, observation or sample unit data.
+        For more info visit https://mermaid-api.readthedocs.io/en/latest/projects.html#project-entity-resources
+        :return: project data.
+        :rtype: dict, None
+        """
+        # Valid data check.
+        if data not in self.proj_resources and data not in self.proj_obs and data not in self.samples:
+            return None
+        # Create path from either name or id parameter.
```

Comment:
Re your question: I don't see a method `_api_projects_path` but think the logic here is sound. The comment is wrong though; this doesn't create a path, it returns the result of an API call. But you could probably lose this comment altogether (and in general I'd tend to take out the more obvious comments).